File: main/src/main/java/lavaplayer/source/youtube/YoutubeSearchMusicResultLoader.java
package lavaplayer.source.youtube;

import lavaplayer.tools.http.ExtendedHttpConfigurable;
import lavaplayer.track.AudioItem;
import lavaplayer.track.AudioTrack;
import lavaplayer.track.AudioTrackInfo;

import java.util.function.Function;

public interface YoutubeSearchMusicResultLoader {
    AudioItem loadSearchMusicResult(String query, Function<AudioTrackInfo, AudioTrack> trackFactory);

    ExtendedHttpConfigurable getHttpConfiguration();
}

File: main/src/main/java/lavaplayer/source/youtube/music/YoutubeSearchMusicResultLoader.kt
package lavaplayer.source.youtube.music

import lavaplayer.tools.http.ExtendedHttpConfigurable
import lavaplayer.track.AudioItem
import lavaplayer.track.AudioTrackFactory

interface YoutubeSearchMusicResultLoader {
    fun loadSearchMusicResult(query: String, trackFactory: AudioTrackFactory): AudioItem

    fun getHttpConfiguration(): ExtendedHttpConfigurable
}
